### PR TITLE
Add recognition by rules

### PR DESCRIPTION
**Pull request by MariiaNebesnova**

### CODE REVIEW

File: src/features/rules/tests/sum.test.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const sum = (a, b) => a + b;
+
+test("adds 1 + 2 to equal 3", () => {
```
Comment (arslanbekova) [inline src/features/rules/tests/sum.test.js:3]: Do we need it?

File: src/features/rules/tests/createSelector.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { createSelector } from "../createSelector";
```
Comment (arslanbekova) [inline src/features/rules/tests/createSelector.test.js:1]: We already have `__tests__` and `__mocks__` directories. Why can't we put these tests there?

File: src/pageServices/contentScripts/findBySelectors.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import { JDNLabel, RulesMap, SelectorsMap } from "../../features/rules/rules.types";
+
+export const findBySelectors = () => {
+  const JDN_HASH = "jdn-hash";
+  const JDN_LABEL = "jdn-label";
+
+  // elements with conflicted labels
+  const conflictLabels: Element[] = [];
+
+  const gen_uuid = () =>
```
[the remark this answers is not in the document]
Comment (MariiaNebesnova) [reply src/pageServices/contentScripts/findBySelectors.ts]: it's not me, it's copy-paste )
fixed

File: src/pageServices/contentScripts/findBySelectors.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        conflictLabels.push(elem);
+        label = `${label} ${jdnLabel}`;
+      }
+      elem.setAttribute(JDN_LABEL, label || jdnLabel);
+      if (!elem.getAttribute(JDN_HASH)) elem.setAttribute(JDN_HASH, gen_uuid());
+    });
+  };
+
+  const cleanUpContent = (elements: Element[], selectorsMap: SelectorsMap) => {
+    const elementsSet = new Set(Array.from(elements));
```
[the remark this answers is not in the document]
Comment (MariiaNebesnova) [reply src/pageServices/contentScripts/findBySelectors.ts]: forgot after refactoring
fixed

File: src/features/rules/tests/sum.test.js (hunk 1 above)
Comment (MariiaNebesnova) [reply src/features/rules/tests/sum.test.js]: no, we don't
removed

File: src/features/rules/tests/createSelector.test.js (hunk 2 above)
Comment (MariiaNebesnova) [reply src/features/rules/tests/createSelector.test.js]: sure, moved to a common folder

File: package.json [hunk 5]
```diff
@@ -49,7 +50,8 @@
     "@sindresorhus/transliterate": "1.6.0",
     "antd": "4.24.3",
     "axios": "0.27.2",
-    "jsdom": "19.0.0",
+    "cyrillic-to-translit-js": "3.1.0",
```
Comment (arslanbekova) [inline package.json:53]: I think that you can update your branch and remove this dependency. We already merged pull request with new translit library

Comment (MariiaNebesnova) [reply package.json]: done

**Review by arslanbekova — APPROVED** [review]

**Pull request merged**